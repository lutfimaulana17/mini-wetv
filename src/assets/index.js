import ImgLogo from './image/logo.png'

import ICSearch from './icon/ic-search.png'
import ICClock from './icon/ic-clock.png'
import ICDownload from './icon/ic-download.png'
import ICNetwork from './icon/ic-network.png'
import ICUser from './icon/ic-user.png'
import ICUpload from './icon/ic-upload.png'


export { ImgLogo, ICSearch, ICClock, ICDownload, ICNetwork, ICUser, ICUpload }