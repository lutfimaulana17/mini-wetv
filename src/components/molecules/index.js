import Header from './Header';
import Footer from './Footer';
import ItemMovie from './ItemMovie';

export { Header, Footer, ItemMovie }