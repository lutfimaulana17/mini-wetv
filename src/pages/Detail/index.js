import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import { ICStar, ICLike } from '../../assets';

import { Header, Footer } from '../../components';

const Detail = () => {
    const [dataMoviePopuler, setDataMoviePopuler] = useState([
        {
            id: 1,
            name: 'Laskar Pelangi',
            img: 'https://images.tokopedia.net/img/cache/700/product-1/2020/7/20/batch-upload/batch-upload_e2df3241-afc4-41e4-b1a3-b4263b2d05e9.jpeg'
        },
        {
            id: 2,
            name: 'Cek Toko Sebelah',
            img: 'https://s3.bukalapak.com/img/3805575796/large/index6.jpg'
        },
        {
            id: 3,
            name: 'Warkop DKI Reborn',
            img: 'https://awsimages.detik.net.id/community/media/visual/2016/08/02/05ec4ce6-6a37-49a2-9910-33722bd206a4.jpg?w=650&q=90'
        }
    ])
    return (
        <div className="detail">
            <Header />
            <div className="detail-video">
                <iframe 
                    width="100%" 
                    height="450px"
                    frameBorder="0" 
                    allow="accelerometer; clipboard-write; encrypted-media; gyroscope; picture-in-picture" 
                    allowfullscreen="allowfullscreen"
                    src="https://www.youtube.com/embed/tgbNymZ7vqY">
                </iframe>
            </div>
            <div className="detail-main">
                <div className="detail-main-left">
                    <div className="detail-main-left-header">
                        <img className="detail-main-left-header-img" src="https://images.tokopedia.net/img/cache/700/product-1/2020/7/20/batch-upload/batch-upload_e2df3241-afc4-41e4-b1a3-b4263b2d05e9.jpeg" />
                        <div className="detail-main-left-header-content">
                            <p className="detail-main-left-header-content-title">Cek Toko Sebelah 2019</p>
                            <p className="detail-main-left-header-content-desc">Release: 1 July 2021</p>
                            <div className="detail-main-left-header-content-rating">
                                <img src={ICStar} className="detail-main-left-header-content-rating-img" />
                                <p className="detail-main-left-header-content-rating-text">7.8</p>
                            </div>
                            <div className="detail-main-left-header-content-favorite">
                                <p className="detail-main-left-header-content-favorite-text">Set Favorite</p>
                                <img src={ICLike} className="detail-main-left-header-content-rating-img" />
                            </div>
                        </div>
                    </div>                
                    <div className="detail-main-left-item">
                        <p className="detail-main-left-item-title">Status</p>
                        <p className="detail-main-left-item-desc">Released</p>
                    </div>
                    <div className="detail-main-left-item">
                        <p className="detail-main-left-item-title">Original Title</p>
                        <p className="detail-main-left-item-desc">Laskar Pelangi 2019</p>
                    </div>
                    <div className="detail-main-left-item">
                        <p className="detail-main-left-item-title">Genre</p>
                        <p className="detail-main-left-item-desc">Animation, Comedy, Adventure, Family.</p>
                    </div>
                    <div className="detail-main-left-item">
                        <p className="detail-main-left-item-title">Overview</p>
                        <p className="detail-main-left-item-desc">Laskar Pelangi (2008) adalah sebuah film garapan sutradara Riri Riza yang dirilis pada 26 September 2008. Film Laskar Pelangi merupakan karya adaptasi dari buku Laskar Pelangi yang ditulis oleh Andrea Hirata. Skenarionya ditulis oleh Salman Aristo yang juga menulis naskah film Ayat-Ayat Cinta dibantu oleh Riri Riza dan Mira Lesmana. Hingga Maret 2009, Laskar Pelangi telah ditonton oleh 4,6 juta orang,[1] menjadikannya film terbanyak ditonton di Indonesia keempat, setelah Jelangkung dengan 5,7 Juta, Pocong 2 dengan 5,1 Juta, dan Ada Apa Dengan Cinta dengan 4,9 Juta.</p>
                    </div>
                    <div className="detail-main-left-item">
                        <p className="detail-main-left-item-title">Production Companies</p>
                        <p className="detail-main-left-item-desc">Dream Works Animation</p>
                    </div>
                    <div className="detail-main-left-item">
                        <p className="detail-main-left-item-title">Production Countries</p>
                        <p className="detail-main-left-item-desc">Indonesia</p>
                    </div>
                </div>
                <div className="detail-main-right">
                    <p className="detail-main-right-title">Film Terpopuler</p>
                    { dataMoviePopuler.map((item) => 
                        <Link to="/detail" style={{ textDecoration: 'none' }}>
                            <div key={item.id} className="detail-main-right-item">
                                <img src={item.img} className="detail-main-right-item-img" />
                                <p className="detail-main-right-item-title">{item.name}</p>
                            </div>
                        </Link>
                    ) }
                </div>
            </div>
            <Footer />
        </div>
    )
}

export default Detail
