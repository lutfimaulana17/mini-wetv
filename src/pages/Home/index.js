import React, { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import "react-responsive-carousel/lib/styles/carousel.min.css";
import { Carousel } from 'react-responsive-carousel';

import { Header, Footer, ItemMovie } from '../../components'
import { clearMovieLatest, clearMovieNowPlaying, clearMoviePopular, clearMovieTopRated, requestMovieLatest, requestMovieNowPlaying, requestMoviePopular, requestMovieTopRated, requestMovieUpcoming, clearMovieUpcoming } from '../../redux/actions/Movie';


const Home = () => {
    const dispatch = useDispatch()
    const stateGlobal = useSelector(state => state)

    const [dataMenu, setDataMenu] = useState([
        {
            id: 1,
            name: 'Semua'
        },
        {
            id: 2,
            name: 'Now Playing'
        },
        {
            id: 3,
            name: 'Popular'
        },
        {
            id: 4,
            name: 'Top Rated'
        },
        {
            id: 5,
            name: 'Upcoming'
        }
    ])
    const [menuActive, setMenuActive] = useState(1)

    useEffect(() => {
        dispatch(requestMovieLatest())
        dispatch(requestMovieNowPlaying())
        dispatch(requestMoviePopular())
        dispatch(requestMovieTopRated())
        dispatch(requestMovieUpcoming())
        return () => {
            dispatch(clearMovieLatest())
            dispatch(clearMovieNowPlaying())
            dispatch(clearMoviePopular())
            dispatch(clearMovieTopRated())
            dispatch(clearMovieUpcoming())
        }
    }, [])

    const { movieNowPlaying, moviePopular, movieTopRated, movieUpcoming } = stateGlobal.Movie;
    return (
        <div className="home">
            <Header />
            <div className="home-wrapper-caurosel">
                <Carousel autoPlay={true} interval={3000} showThumbs={false} showStatus={false}>
                    { moviePopular ?
                        moviePopular.results.length > 0 ?
                            moviePopular.results.slice(0, 5).map((item, index) => 
                                    <Link to={`/detail/${item.id}`} style={{ textDecoration: 'none' }} key={index}>
                                        <div key={item.id}>
                                            <img src={`https://image.tmdb.org/t/p/w1280${item.backdrop_path}`} style={{ height: '450px', objectFit: 'cover' }}/>
                                        </div>
                                    </Link>
                                )
                            : <p>No data!</p>
                        : <p>Loading...</p>
                    }
                </Carousel>
            </div>
            <div className="home-menu">
                { dataMenu.map((item, index) => 
                    <div id={`menu-film${item.id}`} className={item.id === menuActive ? "home-menu-item home-menu-item-active" : "home-menu-item"} key={index} onClick={() => setMenuActive(item.id)}>
                        <p className={item.id === menuActive ? "home-menu-item-text home-menu-item-text-active" : "home-menu-item-text"} >{item.name}</p>
                    </div>
                ) }
            </div>
            <div className="home-main">
                { (menuActive === 1 || menuActive === 2) &&
                    <div className="home-main-section">
                        <p className="home-main-section-title">Now Playing</p>
                        <div className="home-main-section-content">
                            { movieNowPlaying ?
                                movieNowPlaying.results.length > 0 ?
                                    movieNowPlaying.results.slice(0, 10).map((item, index) => 
                                        <ItemMovie id={item.id} poster_path={item.poster_path} title={item.title} key={index} />
                                    )
                                    : <p>No data!</p>
                                : <p>Loading...</p>
                            }
                        </div>
                    </div>
                }
                { (menuActive === 1 || menuActive === 3) &&
                    <div className="home-main-section">
                        <p className="home-main-section-title">Popular</p>
                        <div className="home-main-section-content">
                            { moviePopular ?
                                moviePopular.results.length > 0 ?
                                    moviePopular.results.slice(0, 10).map((item, index) => 
                                        <ItemMovie id={item.id} poster_path={item.poster_path} title={item.title} key={index} />
                                    )
                                    : <p>No data!</p>
                                : <p>Loading...</p>
                            }
                        </div>
                    </div>
                }
                { (menuActive === 1 || menuActive === 4) &&
                    <div className="home-main-section">
                        <p className="home-main-section-title">Top Rated</p>
                        <div className="home-main-section-content">
                            { movieTopRated ?
                                movieTopRated.results.length > 0 ?
                                    movieTopRated.results.slice(0, 10).map((item, index) => 
                                        <ItemMovie id={item.id} poster_path={item.poster_path} title={item.title} key={index} />
                                    )
                                    : <p>No data!</p>
                                : <p>Loading...</p>
                            }
                        </div>
                    </div>
                }
                { (menuActive === 1 || menuActive === 5) &&
                    <div className="home-main-section">
                        <p className="home-main-section-title">Upcoming</p>
                        <div className="home-main-section-content">
                            { movieUpcoming ?
                                movieUpcoming.results.length > 0 ?
                                    movieUpcoming.results.slice(0, 10).map((item, index) => 
                                        <ItemMovie id={item.id} poster_path={item.poster_path} title={item.title} key={index} />
                                    )
                                    : <p>No data!</p>
                                : <p>Loading...</p>
                            }
                        </div>
                    </div>
                }
            </div>
            <Footer />
        </div>
    )
}

export default Home
