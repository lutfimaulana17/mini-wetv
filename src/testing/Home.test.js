import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";

import { Home } from "../pages";
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/store';
import { Footer, Header, ItemMovie } from "../components";

import { ImgLogo } from '../assets'

describe("rendering components home", () => {
    it("renders Home component without crashing", () => {
        shallow(<Provider store={store}>
                    <Router>
                        <Home />
                    </Router>
                </Provider>);
    })

    it("renders Home header without crashing", () => {
        const wrapper = mount(<Router><Header /></Router>)

        const headerLogo = (<img src={ImgLogo} className="header-img" />)
        expect(wrapper.contains(headerLogo)).toEqual(true);

        const headerMenu1 = (<p className="header-menu header-menu-active">Untukmu</p>)
        expect(wrapper.contains(headerMenu1)).toEqual(true);

        const headerMenu2 = (<p className="header-menu">Semua</p>)
        expect(wrapper.contains(headerMenu2)).toEqual(true);
        let defaultLang = wrapper.find('#header-input-lang').props().defaultValue;
        
        expect(defaultLang).toEqual('1');
    })

    it("renders Home item movie without crashing", () => {
        const wrapper = mount(<Router><ItemMovie id="1" title="Testing" poster_path="/iXbWpCkIauBMStSTUT9v4GXvdgH.jpg" /></Router>)

        const title = wrapper.find('.home-main-section-content-item-title').text()
        expect(title).toEqual("Testing")
        expect(wrapper.props().children.props.id).toEqual("1");
        expect(wrapper.props().children.props.poster_path).toEqual("/iXbWpCkIauBMStSTUT9v4GXvdgH.jpg");
        // console.log(wrapper.html())
        // console.log(wrapper.debug())
    })

    it("renders Home footer without crashing", () => {
        const wrapper = mount(<Footer />)

        const textCopyRight = wrapper.find('#text-copyright').text()
        expect(textCopyRight).toEqual('Copyright Â© 1998-2021 Image Future Investment (HK) Limited.');
    })
})

describe("test logic components home", () => {
    const wrapper = mount(
        <Provider store = {store}>
            <Router>
                <Home  />
            </Router>
        </Provider>
    );
    
    it("button category all click - update content", () => {
        wrapper.find("#menu-film1").simulate("click");
        const titleSectionContent = wrapper.find(".home-menu-item-text-active").text();
        expect(titleSectionContent).toEqual('Semua');
    })

    it("button category now playing click - update content", () => {
        wrapper.find("#menu-film2").simulate("click");
        const titleSectionContent = wrapper.find(".home-menu-item-text-active").text();
        expect(titleSectionContent).toEqual('Now Playing');
    })

    it("button category popular click - update content", () => {
        wrapper.find("#menu-film3").simulate("click");
        const titleSectionContent = wrapper.find(".home-menu-item-text-active").text();
        expect(titleSectionContent).toEqual('Popular');
    })
    
    it("button category top rated click - update content", () => {
        wrapper.find("#menu-film4").simulate("click");
        const titleSectionContent = wrapper.find(".home-menu-item-text-active").text();
        expect(titleSectionContent).toEqual('Top Rated');
    })

    it("button category upcoming click - update content", () => {
        wrapper.find("#menu-film5").simulate("click");
        const titleSectionContent = wrapper.find(".home-menu-item-text-active").text();
        expect(titleSectionContent).toEqual('Upcoming');
    })
})

describe("snapshots component Home", () => {
    it("Home snapshots", () => {
      const homeSnap = shallow(<Provider store={store}><Home /></Provider>);
      expect(toJson(homeSnap)).toMatchSnapshot()
    })
  
    it("Home Header snapshots", () => {
        const homeHeaderSnap = shallow(<Header />);
        expect(toJson(homeHeaderSnap)).toMatchSnapshot()
    })

    it("Home Footer snapshots", () => {
        const homeFooterSnap = shallow(<Footer />);
        expect(toJson(homeFooterSnap)).toMatchSnapshot()
    })
})

