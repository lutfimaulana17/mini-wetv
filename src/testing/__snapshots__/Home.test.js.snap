// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots component Home Home Footer snapshots 1`] = `
<div
  className="footer"
>
  <div
    className="footer-left"
  >
    <p
      className="footer-left-text"
      id="text-copyright"
    >
      Copyright Â© 1998-2021 Image Future Investment (HK) Limited.
    </p>
    <p
      className="footer-left-text"
    >
      Ketentuan pelayanan | Perjanjian privasi | Cookie Policy | Saran | Pertanyaan yang sering diajukan | @wetvapp
    </p>
  </div>
  <div
    className="footer-right"
  >
    <img
      className="footer-right-img"
      src="https://puui.qpic.cn/vupload/0/1575530292773_zk8ru850ein.png/0"
    />
    <p
      className="footer-right-text"
    >
      Pindai kode QR untuk mengunduh sekarang!
    </p>
  </div>
</div>
`;

exports[`snapshots component Home Home Header snapshots 1`] = `
<div
  className="header"
>
  <Link
    to="/"
  >
    <img
      className="header-img"
      src="logo.png"
    />
  </Link>
  <p
    className="header-menu header-menu-active"
  >
    Untukmu
  </p>
  <p
    className="header-menu"
  >
    Semua
  </p>
  <div
    className="header-search"
  >
    <input
      className="header-search-input"
    />
    <img
      className="header-search-img"
      src="ic-search.png"
    />
  </div>
  <div
    className="header-download"
  >
    <img
      className="header-download-img"
      src="ic-download.png"
    />
    <div
      className="header-download-desc"
    >
      <p
        className="header-download-desc-text"
      >
        Unduh Aplikasi
      </p>
    </div>
  </div>
  <img
    className="header-icon"
    src="ic-clock.png"
  />
  <img
    className="header-icon"
    src="ic-upload.png"
  />
  <img
    className="header-icon-lang"
    src="ic-network.png"
  />
  <select
    className="header-input-lang"
    defaultValue="1"
    id="header-input-lang"
  >
    <option
      value="1"
    >
      Bahasa Indonesia
    </option>
    <option
      value="2"
    >
      English
    </option>
    <option
      value="3"
    >
      Espanyol
    </option>
    <option
      value="4"
    >
      Portuges
    </option>
  </select>
  <div
    className="header-user"
  >
    <img
      className="header-user-img"
      src="ic-user.png"
    />
  </div>
</div>
`;

exports[`snapshots component Home Home snapshots 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Home />
</ContextProvider>
`;
